src/classes/ADDR_SmartyStreets_Validator.cls

```diff
@@ -222,66 +223,68 @@ public with sharing class ADDR_SmartyStreets_Validator implements ADDR_IValidato
-	                        } else if (sr.zipcodes != null) {
-	                            UTIL_Debug.debug('****No error. Populating raw response field.');
-	                            String response = JSON.serializePretty(sr);
-	                            av.API_Response__c = trimResponse(response);
-	                            av.Verification_Status__c = Label.Addr_Verified;
-	                            
-	                            // if there is at least one city/state, fill in city state
-	                            if (sr.city_states != null && sr.city_states.size() > 0) {
-	                                UTIL_Debug.debug('****Populating city and estate.');
-	                                av.Verified__c = true;
-	                                av.MailingCity__c = sr.city_states[0].city;
-	                                av.MailingState__c = sr.city_states[0].state_abbreviation;
-	                                av.MailingCountry__c = 'US';
-	                            }
-	    
-	                            // if there is exactly one zip, fill in zip
-	                            if (sr.zipcodes.size() == 1)
-	                                av.MailingPostalCode__c = sr.zipcodes[0].zipcode;
-	                        }
-	                    }
-	                } else {
-	                    UTIL_Debug.debug('****Responses did not match number of verifications.');
-	                    for(Address__c av : avList) {
-	                    	av.Verification_Status__c = Label.Zip_Not_Found;
-	                    }
-	                }
-	            }
-	            for (Address__c av : avList) {             
-	                if (av.API_Response__c == null) {   
-	                    av.API_Response__c = Label.Zip_Not_Found;
-	                    av.Verification_Status__c = Label.Zip_Not_Found;
-	                }
-	            }
-	        } catch (Exception ex) {
-	            for (Address__c av : avList) {
-	                UTIL_Debug.debug('****Flagging address as NOT verified Address ID: ' + av.Id 
-	                            + ' - Exception in verifyZipRecords100: ' + ex.getMessage()
-	                            + '\n' + ex.getStackTraceString());
-	                av.Verified__c = false;
-	                av.API_Response__c = ex.getMessage();
-	                av.Verification_Status__c = Label.Zip_Not_Found;
-	                ERR_Handler.processError(ex, ERR_Handler_API.Context.ADDR);
-	            }
-	            ERR_Handler.processError(ex, ERR_Handler_API.Context.ADDR);
-	        }
+            try {
+                ADDR_SmartyStreets_Gateway la = new ADDR_SmartyStreets_Gateway(settings);
+                List<ADDR_SmartyStreets_Gateway.ZipResponse> responses = la.verifyZips(zrList);
+                UTIL_Debug.debug('****Gateway responses: ' + JSON.serializePretty(responses));
+                if (responses != null && !responses.isEmpty()) {
```

Comment:
Why are we checking for !responses.isEmpty() here but not above? It looks like we're doing the empty check already in verifyAddresses and verifyZips.

Reply:
Right, I think we can remove the check since VerifyZips does it already and returns null.